client/state/selectors/has-site-analytics-feature.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import {
+	isEnterprise,
+	isVipPlan,
+	FEATURE_GOOGLE_ANALYTICS,
+	planHasFeature,
+} from '@automattic/calypso-products';
+import isSiteWPCOMOnFreePlan from 'calypso/state/selectors/is-site-wpcom-on-free-plan';
+
+/**
+ * Returns true if the site has the analytics feature, false if the site is WPCOM
+ * and on a free plan or doesn't have this feature.
+ *
+ * @param {object} state Global state tree.
+ * @param {object} site The site to check.
+ * @param {number} siteId The id of the site to check.
+ * @returns {boolean} True if the site has the analytics feature, false otherwise.
+ */
+export default ( state, site, siteId ) => {
```

Comment:
This is messy but I wanted to use the data we already have, rather than re-creating the `site` object in the selector. Let me know if you have any suggestions for improving it.

Reply:
I know we've talked a little about this, but I've just been doing a bit more thinking, and I'm leaning on sticking to using just the `siteId`.

From what I can see of the other selectors we're only ever using the site's ID, so this would make it consistent and I notice that we're using `getSelectedSite` only to pass it to these selectors, so we could get the site object when we need it and default the site ID parameter to the currently selected site - `siteId = getSelectedSiteId( state )` - which seems to be a common pattern.

Reply:
Refactored to use just `siteId` here: https://github.com/Automattic/wp-calypso/pull/53196/commits/4d964cf3942b6bde035d85093b19ee22403dbb6f

I had to the remove tests for no data passed in since `getSelectedSiteId( state )` requires `state`.